### PR TITLE
Improve examples

### PR DESCRIPTION
**Pull request by edongashi**

### CODE REVIEW

File: src/main/metabase/api.ts [hunk 1]
```diff
@@ -406,21 +407,81 @@ export async function getOrCreateTableExamples(tableName: string, aidColumns: st
...
   });
   const dashboardId = dashboard.id as number;
 
   // Pin dashboard.
   await put(`/api/dashboard/${dashboardId}`, { collection_position: 1 });
 
   // Add queries to dashboard.
+
+  const layout = new CardLayout();
+  type CardData = {
```
Comment (pdobacz) [inline src/main/metabase/api.ts:420]: maybe move to `types.ts`?

File: src/main/metabase/card-layout.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+const ROW_WIDTH = 18;
+
+function validateSize(sizeX: number, sizeY: number) {
+  if (sizeX <= 0 || sizeX > ROW_WIDTH || sizeY <= 0) {
+    throw new Error('Invalid rectangle size.');
+  }
+}
+
+export class CardLayout {
```
Comment (pdobacz) [inline src/main/metabase/card-layout.ts:16]: maybe such logic-heavy piece of code could use a unit test?

File: src/main/metabase/examples.ts [hunk 3]
```diff
@@ -1,7 +1,17 @@
...
 import { postgresQuote } from '../../shared';
 import { Field, Table } from './types';
 
-type Display = 'table' | 'bar' | 'row' | 'scalar' | 'map'; // Other types TBD.
+// ----------------------------------------------------------------
+// Types & helpers
+// ----------------------------------------------------------------
+
+/** A section for a group of examples. */
+type ExamplesSection = {
```
Comment (pdobacz) [inline src/main/metabase/examples.ts:10]: similar to other comment - should we maybe move all the types related to metabase API to `types.ts`

File: src/main/metabase/examples.ts [hunk 4]
```diff
@@ -132,35 +220,36 @@ function makeExampleInfos(field: Field, table: Table, aidColumns: string[]): Exa
...
-          name: name,
-          sql: sql,
-          sizeX: 6,
-          sizeY: 4, // TODO: For a table we might need something taller.
-          display: 'table' as Display, // For now we show results only as 'table'.
-          visualizationSettings: {},
-        };
-      }),
+      title: `# Overview of ${table.display_name} columns`,
+      queries: exampleQueries.map(makeQuery),
```
Comment (pdobacz) [inline src/main/metabase/examples.ts:252]: Rename `makeQuery` to `makeCardQuery`, or perhaps rename `exampleQueries` to `exampleInfos`

**Review by pdobacz — APPROVED** [review]
LGTM, some smaller suggestions

File: src/main/metabase/card-layout.ts (hunk 2 above)
Comment (edongashi) [reply src/main/metabase/card-layout.ts:16]: Yep, I've been thinking about it.

File: src/main/metabase/examples.ts (hunk 3 above)
Comment (edongashi) [reply src/main/metabase/examples.ts:10]: This is not exposed to the outer world and is pretty much exclusive to examples, with a reference from a single function in `api`. I don't think it warrants polluting `types`.

File: src/main/metabase/api.ts (hunk 1 above)
Comment (edongashi) [reply src/main/metabase/api.ts:420]: Simplified it to an anonymous type as an Array arg.

File: src/main/metabase/examples.ts (hunk 4 above)
Comment (edongashi) [reply src/main/metabase/examples.ts:252]: The word "Info" vs "Query" is creating confusion. Not sure what naming convention to use...
Maybe we should make everything optional in `ExampleQuery` and provide defaults when materializing the card.

Comment (edongashi) [reply src/main/metabase/examples.ts]: I got rid of `ExampleInfo` entirely. We keep things `undefined` until the last moment. Let me know what you think.

Comment (pdobacz) [reply src/main/metabase/examples.ts]: agreed. I added "info" for lack of better word, to mean something intermediate, from which the API request structure is built. The idea with defaults could work too, let's do whatever is simpler

**Review by pdobacz — APPROVED** [review]
looks good :+1:

**Pull request merged**